From fe6c7e0f06cde65effb3503a47c31ac39aceefb6 Mon Sep 17 00:00:00 2001
From: Alexander Tsoy <alexander@tsoy.me>
Date: Fri, 23 Mar 2018 11:52:27 +0300
Subject: [PATCH 14/14] plymouth: fix detection of plymouth directory
To: <initramfs@vger.kernel.org>

Some distros have both /usr/lib/plymouth and /usr/libexec/plymouth
directorirs, so we should check the existance of plymouth-populate-initrd
script.

Fixes: 421b46f8ae89cfe2b62e880a8a5079ee8c1b3aae
---
 modules.d/50plymouth/module-setup.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/50plymouth/module-setup.sh b/modules.d/50plymouth/module-setup.sh
index 5fbca8b0..b51913e8 100755
--- a/modules.d/50plymouth/module-setup.sh
+++ b/modules.d/50plymouth/module-setup.sh
@@ -6,7 +6,7 @@ pkglib_dir() {
         _dirs+=" /usr/lib/$(dpkg-architecture -qDEB_HOST_MULTIARCH)/plymouth"
     fi
     for _dir in $_dirs; do
-        if [ -d $_dir ]; then
+        if [ -x $_dir/plymouth-populate-initrd ]; then
             echo $_dir
             return
         fi
-- 
2.16.1

